arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -214,6 +229,18 @@ std::pair<ExecutionState, std::unique_ptr<AqlItemBlock>> ExecutionBlockImpl<Exec
   return {state, std::move(result)};
 }
 
+template <class Executor>
+ExecutionState ExecutionBlockImpl<Executor>::getHasMoreState() {
+  if (_done) {
+    return ExecutionState::DONE;
+  }
+  if (_buffer.empty() && _upstreamState == ExecutionState::DONE) {
+    _done = true;
+    return ExecutionState::DONE;
+  }
+  return ExecutionState::HASMORE;
+}
+
 template <class Executor>
 std::pair<ExecutionState, size_t> ExecutionBlockImpl<Executor>::traceSkipSomeEnd(
     ExecutionState state, size_t skipped) {
```

Comment:
Everything I noted above on the `traceGetSome` methods hold for `traceSkipSome` as well. Remove them here and leave them only in the ExecutionBlock.